.github/workflows/tests.yml

```diff
@@ -18,12 +18,24 @@ permissions:
   contents: read
 
 jobs:
+  matrix_prep:
+    runs-on: ubuntu-latest
+    outputs:
+      python_versions_output: ${{ steps.set-matrix.outputs.python_versions }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - id: set-matrix
+        run: |
+          python_versions=$(sed -n "s/^.*Programming Language :: Python :: \([[:digit:]]\+\.[[:digit:]]\+\).*$/'\1', /p" pyproject.toml | tr -d '\n' | sed 's/, $//g')
+          echo "Supported Python versions: $python_versions"
+          echo "python_versions=[$python_versions]" >> "$GITHUB_OUTPUT"
```

Comment:
If you really want to force it, why not to use the list of Python version (as we have in `schedule_tests.yml`) instead of complicated regexp?

Reply:
I wanted to provide a solution that would not need manual modification between `main` and (current, past and future) `stable` branches. Since each `stable` branch can potentially support different set of versions (specifically, `4.2.x` supports `3.8` and `3.9` which `5.0.x` does not support), calculating the matrix automatically from the project setup file felt the correct solution that both saves time not needing manual updates and also not having differences between the `main` and `stable`  branches.

A medium term goal that I also think would be beneficial is to split the Python version matrix building into a reusable component to re use it in multiple workflows if necessary.

Reply:
I can see few issues here, it creates a barrier for many folks in understanding what is going on here. They are force to know regular expressions, `sed`, and to check in another file what kind of versions are supported. IMO, it's unnecessarily complicated.

Reply:
I disagree. The approach of defining the Python versions matrix independently of the branch simplifies maintenance by eliminating the need for manual updates in multiple files.

Considering that maintaining these files falls on the Fellows, I believe it's crucial to streamline processes to support our efforts effectively.

While I acknowledge the concern about complexity as a potential barrier, it's worth noting that these files typically don't receive frequent contributions from the community. Therefore, optimizing them for internal use can enhance efficiency without significant external impact.